src/test/java/io/cdap/plugin/cloud/vision/action/OfflineImageExtractorActionTest.java

```diff
@@ -0,0 +1,371 @@
+  private static final String CROP_HINTS_IMAGE_FILE_PATH = "gs://cloud-samples-data/vision/crop_hints/bubble.jpeg";
+  private static final String FACE_IMAGE_FILE_PATH = "gs://cloud-samples-data/vision/face/faces.jpeg";
+  private static final String PROPERTIES_IMAGE_FILE_PATH =
+    "gs://cloud-samples-data/vision/image_properties/bali.jpeg";
+  private static final String LABELS_IMAGE_FILE_PATH = "gs://cloud-samples-data/vision/label/setagaya.jpeg";
+  private static final String LANDMARKS_IMAGE_FILE_PATH = "gs://cloud-samples-data/vision/landmark/st_basils.jpeg";
+  private static final String LOGOS_IMAGE_FILE_PATH = "gs://cloud-samples-data/vision/logo/google_logo.jpg";
+  private static final String OBJECTS_IMAGE_FILE_PATH = String.format(RESULT_PATH_PATTERN, PATH, OBJECT_IMAGE_NAME);
+  private static final String SAFE_SEARCH_IMAGE_FILE_PATH = String.format(RESULT_PATH_PATTERN, PATH, SAFE_IMAGE_NAME);
+  private static final String TEXT_IMAGE_FILE_PATH = "gs://cloud-samples-data/vision/ocr/sign.jpg";
+  private static final String WEB_DETECTION_IMAGE_FILE_PATH = "gs://cloud-samples-data/vision/web/carnaval.jpeg";
+  private static final String RESULT_FILE_NAME = "output-1-to-1.json";
+
+  private static final String BATCH_SIZE = "20";
+
+  private static Storage storage;
+  private static Bucket bucket;
+
+  @Before
+  public void testSetup() throws Exception {
+    Credentials credentials = CredentialsHelper.getCredentials(SERVICE_ACCOUNT_FILE_PATH);
+    String projectId = CredentialsHelper.getProjectId(PROJECT);
+    storage = getStorage(projectId, credentials);
+
+    GCSPath path = GCSPath.from(PATH);
+    String bucketName = path.getBucket();
+    bucket = storage.get(bucketName);
+    if (bucket != null) {
+      deleteBucket(storage, bucket);
+    }
+
+    BucketInfo bucketInfo = BucketInfo.newBuilder(bucketName)
+      .setStorageClass(StorageClass.STANDARD)
+      .setLocation("us-central1")
+      .build();
+    bucket = storage.create(bucketInfo);
+    initTestsData();
+  }
+
+  @After
+  public void destroy() {
+    deleteBucket(storage, bucket);
+  }
+
+  private static void initTestsData() throws FileNotFoundException {
+    String path = String.format("src/test/resources/%s", OBJECT_IMAGE_NAME);
+
+    FileInputStream serviceAccountStream = new FileInputStream(path);
+    bucket.create(OBJECT_IMAGE_NAME, serviceAccountStream, JPG_CONTENT_TYPE, Bucket.BlobWriteOption.doesNotExist());
+
+    path = String.format("src/test/resources/%s", SAFE_IMAGE_NAME);
+
+    serviceAccountStream = new FileInputStream(path);
+    bucket.create(SAFE_IMAGE_NAME, serviceAccountStream, JPG_CONTENT_TYPE, Bucket.BlobWriteOption.doesNotExist());
+  }
+
+  @Test
+  public void testRunCropHintsFeature() throws Exception {
+    String folder = "crop_hints";
+    String resultFolderPath = String.format(PATH_PATTERN, PATH, folder);
+    OfflineImageExtractorActionConfig config = new OfflineImageExtractorActionConfig(
```

Comment:
Use builder instead to see which properties influence the result, like in `OfflineImageExtractorActionConfigTest `.